\chapter{The \texttt{IsoGene} Package in $R$}
\markchapter{IsoGene R} \label{chap: isogene}

\section{Introduction to \texttt{IsoGene} Package}
\label{sec: intro}

%R is a language and a free software environment for statistical
%computing and graphics. It is a GNU project (a complete Unix-like
%operating system developed in 1984, which is free software) which is
%similar to the S language and environment which was developed at
%Bell Laboratories (formerly AT\&T, now Lucent Technologies) by John
%Chambers and colleagues. R can be considered as a different
%implementation of S. There are some important differences, but much
%code written for S runs unaltered under R.

%$R$ provides a wide variety of statistical and graphical techniques,
%and is highly extensible. It is an integrated suite of software
%facilities for data manipulation, calculation and graphical display.
%$R$, like $S$, is designed around a true computer language, and it
%allows users to add additional functionality by defining new
%functions. $R$ is an environment within which statistical techniques
%are implemented so that it can be extended (easily) via packages.
%The packages supplied with the $R$ distribution and many more are
%available through the Comprehensive R Archive Network (CRAN) family
%of Internet sites covering a very wide range of modern statistics.



%In this chapter, we present an $R$ package called \texttt{IsoGene}
%to implement the statistical methods discussed by Lin \textit{et
%al.}\ (2007), where the primary interest is testing for a monotonic
%relationship between gene-expressions and doses in a microarray
%experiment. %Lin \textit{et al.} discussed the five test statistics,
%including the global likelihood ratio test ($\bar{E}^2_{01}$,
%Bartholomew 1961, Barlow \textit{et al.} 1972, and Robertson
%\textit{et al.} 1988), Williams (1971, 1972), Marcus (1976), M (Hu
%\textit{et al.} 2005) and the modified M (Lin \textit{et al.} 2007)
%and using permutations to obtain the raw $p$-values of the test
%statistics and adjusting $p$-values using BH (Benjamini and Hochberg
%1995) and BY (Benjamini and Yekutilie 2001) procedures controlling
%FDR.


The main IsoGene package functions are \texttt{IsoRawp()} and
\texttt{IsoTestBH()}, which calculate the raw $p$-values using
permutations and adjust them using the BH- and BY-FDR procedures.
The supporting functions \texttt{IsoGene1()} and \texttt{IsoGenem()}
are used to calculate the five test statistics from isotonic
regression for one gene and all the genes, respectively. On the other hand,
the SAM procedure is also implemented to reduce some computational time as
compared to the permutation method. The main function of the SAM is \texttt{IsoTestSAM()},
with supporting functions \texttt{Isofudge(), IsoGenemSAM(), Isoqqstat(), Isoallfdr(), Isoqval()}.
The remaining functions \texttt{IsopvaluePlot(), IsoBHPlot(), IsoSAMPlot()
IsoPlot1()} and \texttt{IsoPlot2()} are used to display the data and
to show the results of testing procedures.

\section{Testing for Trends: Testing Procedures, Multiplicity and Resampling-based Inference}
\subsection{Resampling-based Multiple Testing}

\sloppy{For adjusting for multiple testing, only the BH-FDR procedure
(Benjamini and Hochberg 1995) and BY-FDR procedure (Benjamini and
Yekutieli 2001) are considered in package \texttt{IsoGene()}. The
matrix of the values of the test statistics for each gene and
permutation is referred as the permutation matrix under the null
distribution (see Section 3.2.2).}

% TODO: replace hard-coded sections

This matrix is used to calculate the one-sided $p$-values for the
inference. In the first step the one-sided raw (unadjusted for
multiple testing) $p^{Up}$-values are calculated using (\ref{p1}) or
(\ref{p2}) based on the test statistic $T^{Up}$.

\begin{equation}
P_i=\frac{\#(b: t_{ib} \ge t_{i})}{B-1}, \label{p1}
\end{equation}
where $t_i$ is the observed test statistic for gene $i$.

\begin{equation}
P_i=\frac{\sum_{b=1}^{B} \sum_{j=1}^{m} (t_{jb}\ge
t_{i})}{(B-1) \times m} \label{p2}.
\end{equation}

For $p^{Down}$-values, expect of $\bar{E}^2_{01}$, for which the
test statistic value $t_i$ is always between 0 and 1 and can be
obtained in the same way as $p^{Up}$-values,
\[p^{Down}=\#(b:t_{ib} \le t_{i})/B\; \mbox{or}\; p^{Down}=\sum_{b=1}^{B}
\sum_{j=1}^{m} (t_{jb} \le t_{i})/(B \times m)\] should be used
with $t_{ib}$ and $t_{jb}$ the test statistic values obtained for
gene $i$ and $j$ from permutation $b$. This is because under the
decreasing trend, we reject the four test statistics (namely,
Williams', Marcus', the $M$ and modified $M$) with large negative
values.

Based on the $p$-values, various methods adjusting the type I error
can be applied, such as the Bonferroni, Holm, Hochberg, and BH-FDR
and BY-FDR (Reiner \textit{et al.}\ 2003 and Ge \textit{et al.}\
2003).

\subsection{Significance Analysis of Microarray (SAM)}

SAM (Tusher \textit{et al.}\ 2001, Lin \textit{et al.}\ 2008) is a procedure widely used in the
microarray setting. SAM is a testing procedure, which estimates the
FDR by using permutations under the assumption that all null
hypotheses are true. The procedure consists of three components: (1)
the adjusted test statistics, (2) an approximation of the
distribution of the test statistics based on permutations, and (3)
the control of the FDR.

For a two-group setting, the modified test statistic in SAM is given
by,

\begin{equation}
t_{k}^{SAM}= \frac{\bar{x}_{k}-\bar{x}_{l}} {s_{k}+s_{0}},
\label{ttestSAM}
\end{equation}

where \[ \bar{x}_{l}=\frac{\Sigma_{j=1}^{n_{l}} x_{jl}}{n_{l}},\;\;
\bar{x}_{k}=\frac{\Sigma_{j=1}^{n_{k}} x_{jk}}{n_{k}},\;\;\]
\[s_{k}=\sqrt{\left (\frac{1}{n_{k}}+\frac{1}{n_{l}} \right )
{\frac{\Sigma_{j=1}^{n_{k}}
   (x_{jk}-\bar{x}_{jk})^2+\Sigma_{j=1}^{n_{l}}
   (x_{jl}-\bar{x}_{jl})^2}{n_{k}+n_{l}-2}}},
   \]
and $s_0$ is the fudge factor which is estimated from the data and
is discussed later, $k$ and $l$ are the index of the two groups of
array, and $j$ is the index of the array.

For $F$-type test statistic, such as $\bar{E}_{01}^2$, the modified test
statistic is given by,
\begin{equation}
\bar{E}_{01}^{2SAM}= \frac{\sqrt{\hat{\sigma}^2_{H_0}-\hat{\sigma}^2_{H_1}}} {\sqrt{\hat{\sigma}^2_{H_0}}+s_{0}},
\label{ttestSAM}
\end{equation}






SAM requires that the test statistic for each permutation is sorted
for all the genes, such that the first row of the sorted matrix is
the minimum test statistic across permutations, and the last row is
the maximum, i.e.,
\[\T^{SAM}= \left (
\begin{array}{llll}
t_{(1)1} & t_{(1)2} & \dots & t_{(1)B}\\
t_{(2)1} & t_{(2)2} & \dots & t_{(2)B}\\
.    & .    &.      & .     \\
.    & .    &.      & .     \\
.    & .    &.      & .     \\
t_{(m)1} & t_{(m)2} & \dots & t_{(m)B}\\
\end{array}
\right ).
\]

In $\T^{SAM}$, each element $t_{(i)b}$ is the sorted test statistic
for gene $i$ in permutation $b$. The expected values of the observed
ordered statistics are approximated by the means of the rows of
$\T^{SAM}$, given by $\bar{t}_{(1)}^{SAM}, \bar{t}_{(2)}^{SAM},
\dots, \bar{t}_{(m)}^{SAM}$ that are constructed in the following
way:
\[
\T^{SAM}= \left (
\begin{array}{llll}
t_{(1)1} & t_{(1)2} & \dots & t_{(1)B}\\ \hline t_{(2)1}
& t_{(2)2} & \dots & t_{(2)B}\\ \hline .    & . &.      & .
\\ \hline .    & .    &.      & .     \\ \hline . & .    &.      & .
\\ \hline
t_{(m)1} & t_{(m)2} & \dots & t_{(m)B}\\
\end{array}
\right ) \Rightarrow \left (
\begin{array}{l}
\frac{1}{B}\sum_{b=1}^{B}t_{(1)b} \\
\frac{1}{B}\sum_{b=1}^{B}t_{(2)b} \\
.                                     \\
.                                     \\
.                                     \\
\frac{1}{B}\sum_{b=1}^{B}t_{(m)b} \\
\end{array}
\right ) = \left (
\begin{array}{l}
\bar{t}_{(1)}^{SAM} \\
\bar{t}_{(2)}^{SAM} \\
.                                     \\
.                                     \\
.                                     \\
\bar{t}_{(m)}^{SAM} \\
\end{array}
\right ).
\]

%\noindent \textbf{SAM Procedure}

The SAM procedure proposed by Tusher \textit{et al.}\ (2001) is as
follows:

\begin{enumerate}

\item Compute order statistics $t_{(1)}^{SAM} \le t_{(2)}^{SAM} \le
\dots \le t_{(m)}^{SAM}$.

\item Compute the permutation matrix $\T^{SAM}$.

\item Calculate the expected test statistics
$\bar{t}_{(1)}^{SAM},\bar{t}_{(2)}^{SAM},\dots,\bar{t}_{(m)}^{SAM}$.

\item Plot the $t_{(1)}^{SAM} , t_{(2)}^{SAM}, \dots , t_{(m)}^{SAM}$
values versus the
$\bar{t}_{(1)}^{SAM},\bar{t}_{(2)}^{SAM},\dots,\bar{t}_{(m)}^{SAM}$
values (SAM plot).

\item For a fixed threshold $\Delta$, starting at the origin, and
moving up to the right, find the first $i=i_1$ such that
$t_i^{SAM}-\bar{t}_{i}^{SAM} > \Delta$. All genes, for which
$t_i^{SAM}$ $>$ $t_{i1}^{SAM}$, are called ``significant positive".
Similarly, start at origin, move down to the left and find the first
$i=i_2$ such that $\bar{t}_{2}^{SAM}-t_i^{SAM}> \Delta$. All genes,
for which $t_i^{SAM}$ $<$ $t_{i2}^{SAM}$, are called ``significant
negative". For each $\Delta$ define the upper cut-point
C$_{up}(\Delta)$ as the smallest $t_i^{SAM}$ among the significant
positive genes, and similarly define the lower cut-point
C$_{low}(\Delta)$.

\item For a grid of $\Delta$ values, compute the total number of
significant genes (from step 5), and the median number of falsely
called genes, i.e., the median number of values among each of the
$B$ sets of $t_{ib}$ , $i=1,2,\dots,m$ that fall above cut-point
C$_{up}(\Delta)$ and below cut-point C$_{down}(\Delta)$.
Similarly, compute the $90$th percentile of the number of falsely
called genes.

\item Estimate $\pi_0$, the proportion of truly non-differentially
expressed genes in the data set, as follows:

\begin{enumerate}

\item Compute the first and third quantiles of the permuted
$t^{SAM}$ values, denoted as $q25$ and $q75$ (the $t_i^{SAM}$ are
the values for the original data set; there are $m$ such values).

\item Compute  $\hat{\pi}_0 = \# \{ t_{i} \in (q25, q75)\} / (.5m)$.

\item Let $\hat{\pi}_0 = min (\hat{\pi}_0 , 1)$.
\end{enumerate}

\item  The median and the $90$th percentile of the number of falsely
called genes from step 6, are multiplied by $\hat{\pi}_0$,

\item Pick a $\Delta$ and the corresponding number of significant genes.


\item The FDR is estimated by the median (or the $90$th percentile) of the
number of falsely called genes divided by the number of significant
genes.


%\item $q$-value is the lowest FDR, at which a gene is called
%``significant". It resembles the $p$-value, but is adapted to the
%analysis of a large number of genes. Note that as $ t^{SAM}_i> 0$
%increases, the corresponding $q$-value decreases.

\end{enumerate}


\noindent \textbf{Estimation of the SAM Fudge Factor $s_0$}

In the procedure described above, a fudge factor $s_0$ in the
denominator of the test statistic~(\ref{ttestSAM}) is used. It is
calculated as the percentile of the gene-wise standard error
distribution that minimizes the coefficient of variation (CV) of the
test statistics. This modification is used to overcome bias for
genes with expressions close to zero, which have a large value of
the test statistic due to a small sample variance. By using an
inflated standard error, SAM addresses the problem of the dependence
of the value of the test statistic on the variance of expression
levels for a particular gene. The calculation of $s_0$ is as
follows:

\begin{enumerate}

\item Let $s_{\alpha}$ be the $\alpha \cdot 100\%$ percentile of $s_i$ values.
Let $t_i^{\alpha}= (\bar{X}_{1}-\bar{X}_{0})/(s_i+s_{\alpha})$.

\item Compute the 100 centiles of the $s_i$ values, denoted by $q_1 <
q_2\dots < q_{100}$.

\item For $\alpha \in (0, 0.05, 0.10, \dots, 1.0)$

\begin{enumerate}

\item compute $\nu_j= \mbox{MAD}(t_i^{\alpha}|s_i \in
[q_j,q_{j+1})),j=1,2,\dots,m$, where \mbox{MAD} is the median
absolute deviation from the median, divided by .64;

\item compute $cv(\alpha)=$ coefficient of variation of the $\nu_j$
values.

\end{enumerate}

\item Choose $\hat{\alpha}=argmin[cv(\alpha)]$, i.e.,
$\hat{\alpha}$ is the quantile of the standard error that minimizes
the coefficient of variation of the SAM test statistics.

\item Compute $\hat{s}_0=s_{\hat{\alpha}}$.

\end{enumerate}




\section{Using the \texttt{IsoGene} Package}

\subsection{Data Example}

The data used for the analysis presented below are outcome of a dose-response microarray experiment
consisting of four dose levels. Three microarray samples are available at each dose level (hence, in total
gene expression was measured for 12 arrays). Each array consists of 16,998 genes.

A dataframe with the log2 transformed gene intensities is loaded
into $R$ environment. The first ten genes and first six samples are
displayed, where the row names of the genes show the probe ID,
\texttt{X1, X1.1 and X1.2} are the three arrays for dose zero, while
\texttt{X2, X2.1 and X2.2} are the arrays for the first dose. The dataframe is loaded suing the function \texttt{load()},
\begin{center}
\begin{boxit}
\begin{verbatim}
> load("data.Rdata")
\end{verbatim}
\end{boxit}
\end{center}
A printout of the first 10 lines is given below.
\begin{center}
\begin{boxit}
\begin{verbatim}
> data[1:10,1:6]
                 X1     X1.1     X1.2       X2     X2.1     X2.2
g1 6.923109 7.024719 7.170328 7.219297 7.076908 7.404949
g2 5.107275 5.092935 5.255918 5.312913 4.893855 4.596591
g3 5.913526 6.026197 5.141728 5.828770 5.269202 5.461664
g4   4.919469 4.908159 3.500307 4.814068 4.139949 4.278321
g5   6.002091 5.878718 5.777668 6.214799 5.895586 6.163291
g6 7.162715 7.294693 6.903935 7.223069 6.972928 7.412160
g7   4.049696 4.748409 3.845498 4.780287 4.076589 4.300242
g8   3.191931 4.326571 3.771206 3.570291 2.179324 3.988911
g9   6.487708 6.285804 6.229814 6.109103 6.340837 5.931840
g10   6.695870 6.687039 6.652153 6.503670 6.387794 6.698711
\end{verbatim}
\end{boxit}
\end{center}

\subsection{Loading the Package}

\sloppypar {To load the \texttt{IsoGene} package into $R$, a binary
zip-package of \texttt{IsoGene} program (for Windows) needs to be
installed. \texttt{IsoGene} package requires $R$ packages
\texttt{Multtest} and \texttt{ff}, which need to be installed as
well. Once the packages are installed, they are available for use
after being loaded in memory, which is usually done by the user:}

<<loadPackage>>=
	library(IsoGene)
@

The functions included in the package can be listed using the $R$
help system:

<<eval=FALSE>>=
	help(package = IsoGene)
@

%\begin{figure}[!h]
%\centering
%{\subfigure{\resizebox{.89\textwidth}{!}{\includegraphics{D:/Data/Ph.Dthesis/Figures/IsoGeneLib/page1.eps}}}\\
%\subfigure{\resizebox{.9\textwidth}{!}{\includegraphics{D:/Data/Ph.Dthesis/Figures/IsoGeneLib/page2.eps}}}
%} \caption{\em{The main help file of IsoGene package.}}
%\label{IsoGeneHelp}
%\end{figure}



%\begin{figure}[!h]
%\centering
%{\includegraphics[width=.6\textwidth]{IsoGenehelp.eps}}
%\caption{\em{The main help file of IsoGene package.}}
%\label{IsoGeneHelp}
%\end{figure}


%The help file of \texttt{IsoGene} package is shown in
%Figure~\ref{IsoGeneHelp}.
First, \texttt{IsoPlot1()} and \texttt{IsoPlot2()} can be used to
explore the data. Second, \texttt{IsoGene1()} and
\texttt{IsoGenem()} can be used to calculate the test statistics.
Third, \texttt{IsoRawp()} provides the output for two-sided or
one-sided $p$-values ($p^{Up}$ or $p^{Down}$). Based on the
$p$-values obtained, one can choose one test statistic and
multiplicity adjustment method for inference by using
\texttt{IsoTestBH()}. Finally, \texttt{IsopvaluePlot()} can be
useful for examining both of $p^{Up}$- or $p^{Down}$-values, and in
particularly, as a post hoc procedure it can be used to examine
genes with both small $p^{Up}$- and $p^{Down}$-values.


\section{The \texttt{IsoGene} Functions}
\subsection{Quick Start}
The first stage of the analysis (which is also the time consuming stage) consists of permutations under the null hypothesis in order to obtain the distribution of the test statistic under the null hypothesis. Note that, by default, all five test statistics discussed above are calculated. The function \texttt{IsoRawp()} is used to perform the permutation. A general call of the function \texttt{IsoRawp()} has the form of
\begin{center}
\begin{boxit}
\begin{verbatim}
IsoRawp(x, data, niter=1000, seed=1234)
\end{verbatim}
\end{boxit}
\end{center}
Here, \texttt{x} is a vector which contains the dose levels and \texttt{data} is the R object, which contains the information about gene expression and genes names. Once the permutation stage is completed, the FDR adjusted $p$-values can be obtained using function \texttt{IsoTestBH()}. The function calculates the adjusted p values for each statistic using either the BH-FDR or BY-FDR for multiplicity adjustment. The user can specify one of the five test statistics discussed above, or use the default call, in the later case adjusted $p$-values for all test statistics will be calculated. A general form of the \texttt{IsoTestBH()} has the form
\begin{center}
\begin{boxit}
\begin{verbatim}
IsoTestBH(rp, FDR=c(0.05,0.1), type=c("BH","BY"),
stat=c("E2","Williams","Marcus","M","ModifM"))
\end{verbatim}
\end{boxit}
\end{center}
Note that \texttt{rp} is an R object, which contains all the output produced by the function \texttt{IsoRawp()}.\newline In what follows we illustrate in more details the use of the functions of the \texttt{IsoGene} package.

\subsection{Exploring the Data}

\texttt{IsoPlot1()} and \texttt{IsoPlot2()} are two plotting
functions that can be used to explore the data. Scatterplots for the
second gene in the dataset (\texttt{data[2,]}) can be produced by

<<IsoPlot>>=
  x <- c(rep(1,3), rep(2,3), rep(3,3), rep(4,3))
  gene1 <- data[2,]
  par(mfrow = c(1,2))
  IsoPlot1(x, y = gene1)
  IsoPlot2(x, y = gene1)	
@
 

\begin{figure}[!h]
\centering
{\includegraphics[width=.6\textwidth]{exgene.eps}}
\caption{\em{The data points are plotted as circles, while sample
means as pluses. The right panel additionally plots the fitted
increasing isotonic regression model (blue solid line).}}
\label{exgene}
\end{figure}


The left panel in Figure~\ref{exgene} shows the original data points
(as circles) and sample means (as pluses) for each dose. The right
panel in Figure~\ref{exgene} shows the increasing isotonic
regression model (blue solid line) fitted on the data. The fitted
monotonic line does not indicate the significance of the test, but
simply shows a more likely increasing (or decreasing) trend.


\subsection{Calculating the Test Statistics}

The five test statistics described in Chapter 2 can be obtained by
using the function \texttt{IsoGene1()} for a single gene and using
the function \texttt{IsoGenem()} for all the genes simultaneously.
The following $R$ codes illustrate the input and output generated by
these two functions:

<<IsoGene1>>=
	stat1 <- IsoGene1(x, gene1)
@

The object \texttt{stat1} contains the information about the five test statistics and the direction for which the likelihood is maximizes.

<<stat1>>=
	stat1
@

The first 10 objects are the values calculated for the five test
statistics under increasing and decreasing trends. The last object
indicates the higher likelihood of isotonic regression with ``u"
meaning a increasing trend or ``d" meaning a decreasing trend.


We use the first 10 genes as an example to illustrate the use of
function \texttt{IsoGenem()}:

<<IsoGenem>>=
	statm <- IsoGenem(x, data[1:10,])
  statm
@


The output from \texttt{IsoGenem()} has the same structure as the one
for \texttt{IsoGene1()}, but each object contains the values of the
test statistics and the likely direction of the isotonic regression
model for all the genes.


\subsection{Obtaining Raw $p$-values}

As discussed above, we use permutations to obtain the raw $p$-values
for the five test statistics. The function \texttt{IsoRawp()} can be
used in the following way:

<<IsoRawp>>=
  rawp <- IsoRawp(x=x, y=data, niter=1000, seed=1234)
@


The four arguments in this function need to be specified, with no
default pre-specified values. \texttt{x} is the explanatory variable
indicating the dose levels for all the samples in the data.
\texttt{data} is the data frame of the gene expression.
\texttt{niter} defines the number of permutations used to
approximate the null distribution and \texttt{seed} determines the
random seed used to generate the permutations. The output item
\texttt{rawp} contains four objects with $p$-values for the five
test statistics: the first one contains the two-sided $p$-values,
the second contains the one-sided $p$-values, the third contains
$p^{Up}$-values, and the last one contains $p^{Down}$-values. Below
we print a part of the object with two-sided $p$-values for illustration:

<<rawptwosided>>=
	rawp.twosided <- rawp[[1]]
@


The first 10 rows in of the object \texttt{rawp.twosided} are
<<>>=
	rawp.twosided[1:10,]
@


The first output object from \texttt{rawp} is a matrix with six
columns, where the first column indicates the probe ID. Columns from
the second to the sixth are $p$-values for each of the five test
statistics, respectively. The remaining three output objects
(\texttt{rawp}[[2]], \texttt{rawp}[[3]], \texttt{rawp}[[4]]) are
structured in the same way.


\subsection{Plot of $p$-values for a Single Gene}

For a single gene, the function \texttt{IsopvaluePlot()} can be used
to show the $p^{Up}$ and $p^{Down}$-values for a given test
statistic:

<<eval=FALSE>>=
  IsopvaluePlot(x, y, niter, seed, stat = c("E2", "Williams",
"Marcus", "M", "ModifM"))
@


We use one gene as an example to illustrate how $p^{Up}$ and
$p^{Down}$-values (in the upper and lower panels of
Figure~\ref{expvalue}) are obtained. In Figure~\ref{expvalue}, the
observed test statistics are drawn as the dashed line, and the
values of the test statistics obtained from permutations are spread
over the x-axis. For this gene, the $p^{Up}$ is much smaller as
compared to the $p^{Down}$ since $T^{Up} \gg T^{Down}$, which
implies a possible increasing trend in the data.

<<IsopvaluePlot, fig=TRUE, include=FALSE>>=
	gene1 <- data[2,]
  IsopvaluePlot(x, gene1, niter=1000, seed=123, stat="E2")
@

  
\begin{figure}[!h]
\centering
{\includegraphics[height=10cm,width=12cm]{IsoGene-IsopvaluePlot}}
\caption{\em{The $p^{Up}$ and $p^{Down}$-values using
$\bar{E}_{01}^2$ for an example gene. The dashed line is the
observed test statistic value. In the upper panel, the dashed line
(at the right) is larger than most of the test statistics from
permutations, which results in a small $p^{Up}$-value. In the lower
panel, the dashed line (close to zero) is smaller than most of the
test statistics from permutations, which results in a large
$p^{Down}$-value.}} \label{expvalue}
\end{figure}



\subsection{BH/BY-FDR Procedures for Adjusting for Multiple Testing}

With the two-sided $p$-values, the user needs to select one of the
five test statistics, the FDR level, and the type of multiplicity
adjustment (BH-FDR or BY-FDR) to obtain the list of significant
genes:

<<IsoTestBH>>=
  IsoTestBH(rp, FDR = c(0.05,0.1), type = c("BH","BY"),
    stat=c("E2","Williams","Marcus","M","ModifM"))
@


The following example shows the use of the global likelihood ratio
test $\bar{E}^{2}_{01}$, the FDR level of 0.05 and the BH-FDR
procedure controlling the FDR: \\ \\

<<E2BH>>=
  E2.BH <- IsoTestBH(rawp.twosided, FDR=0.05, type="BH", stat="E2")
@

The first 10 rows in the object \texttt{E2.BH} list the sorted row and adjusted p values for the $\bar{E}^{2}_{01}$ statistic.

<<E2BHrows>>=
  E2.BH[1:10,]	
@
%s

Here we show only the first ten genes declared significant by using
$\bar{E}^{2}_{01}$ test. The output results in a matrix of five
columns: the first column indicates the probe ID, the second column
is the corresponding row number of significant genes in the original
dataset, the third column is the unadjusted/raw $p$-value, and the
last column is the adjusted $p$-value using the requested ``BH"
procedure. The order of the list of genes found significant is based
on the row number. Moreover, the function \texttt{IsoBHPlot()} can
be used to visualize the number of significant findings for the
BH-FDR and BY-FDR procedures for the specified test statistic:

<<IsoBHPlotExample,eval=FALSE>>=
  IsoBHPlot(rp, FDR = c(0.05,0.1),
    stat=c("E2","Williams","Marcus","M","ModifM"))
@

Figure~\ref{IsoBHPlot} shows the unadjusted (solid blue line) and
the BH-FDR (dotted and dashed red line) and BY-FDR (dashed green
line) adjusted $p$-values for $\bar{E}^2_{01}$. It is obtained using
the function \texttt{IsoBHPlot()}:

<<IsoBHPlot,fig=TRUE, include=FALSE>>=
	IsoBHPlot(rawp.twosided, FDR=0.05, stat="E2")
@

\begin{figure}[!h]
\centering
{\includegraphics[width=.6\textwidth]{IsoGene-IsoBHPlot}}
\caption{\em {The unadjusted (solid blue line), and the BH-FDR
(dotted and dashed red line) and BY-FDR (dashed green line) adjusted
$p$-values for $\bar{E}_{01}^2$.}} \label{IsoBHPlot}
\end{figure}



\section{Significance Analysis of Dose-response Microarray Data (SAM)}

In this package, we also implement the significance analysis of microarray (SAM) for testing
for the dose-response relationship under order restricted alternatives. The SAM procedure was proposed
by Tusher \textit{et al.}\ (2001) for finding differentially expressed genes by using permutations while controlling for
the FDR.

The main function for the SAM procedure is \texttt{IsoTestSAM()}. Within the main function, \texttt{Isofudge()}
calculates the fudge factor in the SAM test statistic, \texttt{IsoGenemSAM()} is used
to obtain the values of SAM test statistics, \texttt{Isoqqstat()} calculates the SAM test statistic
for the required number of permutations specified by users, \texttt{Isoallfdr()} obtains the delta table in the SAM procedure,
\texttt{Isoqval()} computes q-values of the SAM.  %And we can use \texttt{IsoSAMPlot}() to display outputs of the SAM.

The syntax of these function is as follows,

<<IsoSyntax,eval=FALSE>>=
  Isofudge(x, y)
  IsoGenemSAM(x, y, fudge.factor)
  Isoqqstat(x, y, fudge=c(0,"pooled"), niter=100)
  Isoallfdr(qqstat, ddelta, stat = c("E2","Williams","Marcus","M","ModifM"))
  Isoqval(delta, allfdr, qqstat, stat)	
@

We use the same data as above to obtain the fudge factor of the SAM procedure for
the five test statistics by using function \texttt{Isofudge()}.

<<fudgeFactor>>=
  fudge.factor <- Isofudge(x, data)
@

The output of this function gives a vector of five fudge factors for each of the test statistics.

<<fudgeFactorOutput>>=
  fudge.factor
@

Note that the fudge factor of the $\bar{E}_{01}^2$ is obtained based on the algorithm for F test
statistics given by Tusher \textit{et al.}\ (2001) and should be used with cautions. The performance of using
the fudge factor as compared to the $t$-type test statistics has not yet investigated in term of
power and control of the FDR. Therefore, it's advisable to use the fudge factor in the $t$-type test statistics.

<<SAMtest>>=
  SAMtest.stat <- IsoGenemSAM(x, data, fudge.factor)
@

The output of the function gives

<<namesSAMtest>>=
	names(SAMtest.stat)
@

The following codes produce the values of the five test statistics for the first ten genes,

<<SAMtestStat>>=
  SAMtest.stat[[1]][1:10]
  SAMtest.stat[[2]][1:10]
  SAMtest.stat[[3]][1:10]
  SAMtest.stat[[4]][1:10]
  SAMtest.stat[[5]][1:10]
  SAMtest.stat[[6]][1:10]
@


To obtain the SAM test statistics for one of five test statistic values, for example, the modified M test,
with the required number of permutations specified by users and compute the
delta table in the SAM procedure, we can use function \texttt{Isoqqstat()} and \texttt{Isoallfdr()} as follows,

<<qqstat>>=
	qqstat <- Isoqqstat(x, data, fudge="pooled", niter=100)
  dtable <-  Isoallfdr(qqstat, , stat="ModifM")
  head(dtable, n = 4)
  # TODO: identify row with largest FDR < 0.05
  tail(dtable, n = 4)
@

  
Note that in \texttt{Isoallfdr()}, ddelta is left blank, with default value taken
from the data, i.e., all the percentiles of the standard errors.
By fixing the 50\% FDR at 0.05, the corresponding delta value is 0.83
(marked in-between the dashed lines) as we obtain from the delta table above,
the number of differentially expressed genes are 872 with potential 42 genes as false positives.

<<qvalOutput>>=
  qval <- Isoqval(delta=0.83, allfdr=dtable, qqstat=qqstat,
    stat="ModifM")
  dim(qval[[1]])
  head(qval[[1]])
  tail(qval[[1]])
  dim(qval[[2]])
  head(qval[[2]])
  tail(qval[[2]])	
@

By specifying the desired delta value, delta table, and the user-defined test statistic in function \texttt{Isoqval()}, we can obtain
the q value of each gene from the SAM procedure. The first object of the output is the list of q values for
all the genes, ranking from the smallest test statistic value to the largest; while the second object is the list of q values
for the 872 differentially expressed genes, ranking from the smallest test statistic value to the largest. The first column of the output matrices is the row number of genes in
the data set, the second column is the observed modified M test statistic value, and the last column is the q value of the
SAM procedure for both objects.

Alternatively, we can use function \texttt{IsoTestSAM()} to summarize all the steps above and give results of a list of significant findings, which is the same second output of function \texttt{Isoqval()}.

<<IsoTestSAM, eval=FALSE>>=
  IsoTestSAM(x, y=data, fudge=c(0,"pooled"), niter=100, 
    seed=123, FDR=0.05, stat=c("E2","Williams","Marcus","M","ModifM"))
@

Specifying the same options as above in this function, we can obtain the list of significant genes as follows: the first column is the Probe.ID, the second column is the corresponding row numbers of the genes in the data set, the third column is the observed modified M SAM test statistics, the fourth column is the $q$ values of genes by using the SAM procedure. The last two columns gives additional information by calculating the $p$ values based on the SAM permutation matrix and adjusting these $p$ values using the BH-FDR procedure.

<<IsoSAMobj>>=
  IsoSAM.obj <- IsoTestSAM (x, y = data, fudge = "pooled", 
    niter = 100, FDR = 0.05, stat = "ModifM")
  head(IsoSAM.obj)
  tail(IsoSAM.obj)
@

Finally, the graphic output of the SAM procedure can be produced using function \texttt{IsoSAMPlot()}.

<<IsoSAMPlot,eval=FALSE>>=
	IsoSAMPlot(qqstat, allfdr, FDR=0.05, stat=c("E2","Williams",
    "Marcus","M","ModifM"))
@

This function requires the use of output from \texttt{Isoqqstat()} and \texttt{Isoallfdr()}, given a user-defined test statistic
and the FDR level to control. We still take the modified M test statistic for example, at the FDR of 0.05. There are four plots
yielded from the SAM procedure. Panel $a$ shows the FDR (either 50\% or 90\% (more stringent)) vs. $\Delta$, from which, user can choose the delta value with
the corresponding desired FDR. Panel $b$ shows the number of significant genes vs. $\Delta$, and panel $c$ shows the number of false positives (either 50\% or 90\%) vs. $\Delta$. Finally panel $d$ shows the observed vs. the expected (obtained from permutations) test statistics, in which the red dots are those genes called differentially expressed.

<<IsoSAMPlotOutput, fig=TRUE, include = FALSE>>=
	IsoSAMPlot(qqstat = qqstat, allfdr = dtable, FDR = 0.05, stat = "ModifM")
@



\begin{figure}[!h]
\centering
{\includegraphics[width=.7\textwidth]{IsoGene-IsoSAMPlotOutput}}
\caption{\em {The SAM plots: a.Plot of the FDR vs. delta; b. Plot of number of significant genes vs. delta; c. Plot of
number of false positives vs. delta; d. Plot of observed vs. expected test statistics.}} \label{IsoBHPlot}
\end{figure}



\clearpage
\newpage


\section*{References}


\begin{list}{}{\setlength{\leftmargin}{.3in}\setlength{\itemindent}{-.2in}}

\item Agresti, A. (1997) \textit{Statistical Methods for the
Social Sciences}, Finlay.

\item Barlow, R.E., Bartholomew, D.J.,
Bremner, M.J. and Brunk, H.D. (1972) \textit{Statistical inference
under order restriction}, New York: Wiley.

\item Bartholomew, D.J. (1961) Ordered
tests in the analysis of variance, \textit{Biometrika}, \textbf{48},
325--332.%

\item Benjamini, Y. and Hochberg, Y.
(1995) Controlling the false discovery rate: a practical and
powerful approach to multiple testing. \textit{Journal of Royal
Statistical Soceity, Biostatistics}, \textbf{57}, 289--300.%


\item Benjamini, Y. and Yekutieli,
D. (2001) The control of the false discovery rate in multiple
testing under dependency. \textit{Annal of Statistics},
\textbf{29(4)}, 1165--1188.%


\item
Benjamini, Y. and Yekutieli, D. (2005) False Discovery
Rate-Adjusted Multiple Confidence Intervals for Selected Parameters.
{\em Journal of the American Statistical Association}, \textbf{100}, 71--81.%


\item Chuang-Stein, C. and Agresti, A. (1997) Tutorial in biostatistics:
A review of tests for detecting a monotone dose-response
relationship with ordinal response data, \textit{Statistics in
Medicine}, \textbf{16}, 2599--2618.%


\item Ge, Y., Dudoit, S. and Speed, P.T.
(2003) Resampling based multiple testing for microarray data
analysis,  \textit{technical report}, \textbf{633}, University of
Berkeley.%

\item Hu, J., Kapoor, M., Zhang, W.,
Hamilton, S.R. and Coombes, K.R. (2005) Analysis of dose response
effects on gene expression data with comparison of two microarray
platforms, \textit{Bioinformatics}, \textbf{21(17)}, 3524--3529.%

\item Lin, D., Shkedy, Z., Yekutieli, D., Burzykowki,
T., G\"{o}hlmann, H.W.H., De Bondt, A., Perera, T., Geerts, T.,
Bijnens, L. (2007a) Testing for trend in dose-response microarray
experiments: comparison of several testing procedures, multiplicity,
and resampling-based inference. \textit{Statistical Application in
Genetics and Molecular Biology}, \textbf{6(1)}, article 26.%

\item Lin, D., Shkedy, Z., Burzykowki, R., Ion,
T., G\"{o}hlmann, H.W.H., De Bondt, A., Perera, T., Geerts, T.,
Bijnens, L. (2008) An investigation on performance of Significance
Analysis of Microarray (SAM) for the comparisons of several
treatments with one control in the presence of small variance genes.
\textit{Biometrical Journal}, Multiple Comparison Problem, Special Issue,
\textbf{50(5)}, 801--823.%



\item Marcus, R. (1976) The powers of some tests of
the quality of normal means against an ordered alternative,
\textit{Biometrika}, \textbf{63}, 177--83.%

\item Reiner, A., Yekutieli, D. and
Benjamini, Y. (2003) Identifying differentially expressed genes
using false discovery rate controlling procedures.
\textit{Bioinformatics}, \textbf{19(3)}, 368--375.%

\item Robertson, T., Wright, F.T. and Dykstra, R.L. (1988) \textit{ Order restricted statistical inference}, Wiley.%



item Ruberg, S.J. (1995a) Dose response studies. I. Some design
considerations. \textit{Journal of Biopharmaceutical Statistics},
\textbf{5(1)}, 1--14.%

\item Ruberg, S.J. (1995b) Dose response studies. II. Analysis and
interpretation. \textit{Journal of Biopharmaceutical Statistics},
\textbf{5(1)}, 15--42.%

\item Tusher, V.G., Tibshirani, R. and
Chu, G. (2001) Significance analysis of microarrys applied to the
ionizing radiation response, \textit{Proceedings of the National
Academy of Sciences}, \textbf{98} , 5116--5121.%


\item Williams, D.A. (1971) A test for differences between treatment
means when several dose levels are compared with a zero dose
control, \textit{Biometrics}, \textbf{27}, 103--117.%

\item Williams, D.A. (1972) The comparision of several dose levels
with a zero dose control, \textit{Biometrics}, \textbf{28},
519--531.%


\end{list}




